autoload/airline/extensions/projectdir.vim

```diff
@@ -0,0 +1,18 @@
+" MIT License. Copyright (c) 2017-2021 C.Brabandt et al.
+" Plugin: https://github.com/tpope/vim-fugitive
+" vim: et ts=2 sts=2 sw=2
+
+scriptencoding utf-8
+
+if !airline#util#has_fugitive()
+  finish
+endif
+
+function! airline#extensions#projectdir#Dir() abort
+  let dir = FugitiveGitDir(bufnr(''))
+  return empty(dir) ? '' : pathshorten(fnamemodify(simplify(dir .. '/../'), ':h:t'))
```

Comment:
This will return the wrong result for submodules, multiple work trees, bare repositories, and probably other things. Recommended method to retrieve the work tree is `FugitiveFind(':/', bufnr(''))`.